bot/cogs/team.py

```diff
@@ -0,0 +1,346 @@
+from random import shuffle
+
+from discord import Member, VoiceChannel, errors
+from discord.ext import commands, tasks
+from common.discord_helper import get_member_list, string_to_role
+
+from common.bot_logger import get_logger
+from common.helper_functions import reply_and_delete
+
+logger = get_logger(__name__)
+
+
+class Team:
+    def __init__(self, name: str):
+        self.__name = name
+        self.__members = []
+        self.__size = 0
+        self.__vc = None
+
+    @property
+    def name(self):
+        return self.__name
+
+    @name.setter
+    def name(self, name: str):
+        self.__name = name
+
+    @property
+    def size(self):
+        return self.__size
+
+    @property
+    def members(self):
+        return self.__members
+
+    def add_member(self, member: Member):
+        if member not in self.members:
+            self.__members.append(member)
+            self.__size += 1
+
+    def rem_member(self, member: Member):
+        for m in self.__members:
+            if m.id == member.id:
+                self.__members.remove(m)
+                self.__size -= 1
+
+    @property
+    def vc(self):
+        return self.__vc
+
+    @vc.setter
+    def vc(self, vc: VoiceChannel):
+        self.__vc = vc
+
+
+class TeamCog(commands.Cog, name="Team commands"):
+    def __init__(self, bot):
+        self.bot = bot
+        self.teams = []
```

Comment:
This might result in an issue when the bot is connected to multiple guild, any suggestions are welcome